src/libs/ajax/data-table-providers/WdsDataTableProvider.ts

```diff
@@ -136,20 +140,24 @@ export class WdsDataTableProvider implements DataTableProvider {
   }
 
   // transforms a WDS array to Entity Service array format
-  private transformAttributes = (attributes: RecordAttributes): RecordAttributes => {
-    return _.mapValues(val => {
+  protected transformAttributes = (attributes: RecordAttributes, primaryKey: string): RecordAttributes => {
+    // the pickBy here excludes the primary-key attribute from the attribute map. The data table reads
+    // the primary-key attribute name from the idName of entity metadata, and its value from the entity name.
+    // if we included the primary-key attribute in the attribute map, the data table would show a duplicate column.
+    return _.pickBy((_v, k) => k !== primaryKey, _.mapValues(val => {
       return _.isArray(val) ? this.toEntityServiceArray(val) : this.maybeTransformRelation(val)
-    }, attributes)
+    }, attributes))
   }
 
-  protected transformPage = (wdsPage: RecordQueryResponse, recordType: string, queryOptions: EntityQueryOptions): EntityQueryResponse => {
+  protected transformPage = (wdsPage: RecordQueryResponse, recordType: string, queryOptions: EntityQueryOptions, metadata: EntityMetadata): EntityQueryResponse => {
     // translate WDS to Entity Service
     const filteredCount = wdsPage.totalRecords
     const unfilteredCount = wdsPage.totalRecords
+    const primaryKey = !!metadata[recordType] ? metadata[recordType].idName : 'sys_name' // for safety; recordType should always be present
```

Comment:
If `sys_name` was the result here, would this be somewhat of a silent failure though?

Reply:
it would, yeah. Which would result in a potentially incorrect label for the first column of the table - but I thought that was preferable to a JS error that breaks the page.

Reply:
I've updated `sys_name` -> `(unknown column)` to try to make this a little clearer.